packages/shared_preferences/shared_preferences_platform_interface/lib/better_shared_preferences_platform_interface.dart

```diff
@@ -0,0 +1,252 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'better_method_channel_shared_preferences.dart';
+import 'better_types.dart';
+
+/// The interface that implementations of shared_preferences must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `shared_preferences`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [BetterSharedPreferencesPlatform] methods.
+abstract base class BetterSharedPreferencesPlatform {
```

Comment:
Do we need a completely new platform interface, or can we just add new methods to the old one? What are the pros/cons of a full split?

Reply:
I'm hoping to keep them separate so that deprecation is simpler. I guess it may not matter in the end. I wrote most of the code as it's own file (here and on the platforms) so that people could get a better feeling for the new systems as they are without the old code dirtying the water.

I'm open to merging files if you think that's best.